index.bs

```diff
@@ -1659,96 +1708,145 @@ session to exist at any particular time, or by providing more complicated sessio
+        and cryptographic parameters represented by the first [=list/item=] in |credTypesAndPubKeyAlgs| that is supported by
+        this authenticator.
+    1. Let |userHandle| be <code>|userEntity|.{{PublicKeyCredentialUserEntity/id}}</code>.
+    1. Let |credentialSource| be a new [=public key credential source=] with the fields:
+        <dl link-for="public key credential source">
+            : [=type=]
+            :: {{PublicKeyCredentialType/public-key}}.
+            : [=privateKey=]
+            :: |privateKey|
+            : [=rpId=]
+            :: |rpId|
+            : [=userHandle=]
+            :: |userHandle|
+            : [=otherUI=]
+            :: Any other information the authenticator chooses to include.
+        </dl>
+    1. If |requireResidentKey| is true or the authenticator chooses to create a [=Client-side-resident Credential Private Key=]:
+        1. Let |credentialId| be a new [=credential id=].
+        1. Set |credentialSource|.[=public key credential source/id=] to |credentialId|.
+        1. Let |credentials| be this authenticator's [=credentials map=].
+        1. [=map/Set=] |credentials|[(|rpId|, |userHandle|)] to |credentialSource|.
+    1. Otherwise:
+        1. Let |credentialId| be the result of serializing and encrypting |credentialSource| so that only this authenticator can
+            decrypt it.
 1. If any error occurred while creating the new credential object, return an error code equivalent to "{{UnknownError}}" and
     terminate the operation.
-1. Process all the supported extensions requested by the client, and generate the [=authenticator data=] with
-    [=attestation data=] as specified in [[#sec-authenticator-data]]. Use this [=authenticator data=] and the
-    [=hash of the serialized client data=] to create an [=attestation object=] for the new credential using the procedure
-    specified in [[#generating-an-attestation-object]]. For more details on attestation, see [[#sctn-attestation]].
+1. Let |processedExtensions| be the result of [=authenticator extension processing=] for each supported extension identifier/input
+    pair in |extensions|.
+1. Let |attestationData| be the [=attestation data=] byte array including the |credentialId| and |publicKey|.
+1. Let |authenticatorData| be the byte array specified in [[#sec-authenticator-data]] including |attestationData| and any |processedExtensions|.
+1. Return the [=attestation object=] for the new credential created by the procedure specified in
+    [[#generating-an-attestation-object]] using an authenticator-chosen [=attestation statement format=], |authenticatorData|,
+    and |hash|. For more details on attestation, see [[#sctn-attestation]].
 
 On successful completion of this operation, the authenticator returns the [=attestation object=] to the client.
 
 
-### The <dfn>authenticatorGetAssertion</dfn> operation ### {#op-get-assertion}
+<h4 id="op-get-assertion" algorithm> The <dfn>authenticatorGetAssertion</dfn> operation</h4>
```

Comment:
should be renamed to authenticatorGetCredential (if we stick to the redefinition of credential)